services/factorio-mod-portal/factorio-mod-portal.service.js

```diff
@@ -0,0 +1,198 @@
+    {
+      title: 'Factorio Mod Portal mod version',
+      namedParams: { modName: 'rso-mod' },
+      staticPreview: this.render({ version: '6.2.20' }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'latest version' }
+
+  static render({ version }) {
+    return renderVersionBadge({ version })
+  }
+
+  async handle({ modName }) {
+    const { latest_release } = await this.fetch({ modName })
+    return this.constructor.render({ version: latest_release.version })
+  }
+}
+
+// Badge for mod's latest compatible Factorio version
+// Query 'range' to display range of compatible versions
+class FactorioModPortalFactorioVersions extends BaseFactorioModPortalService {
+  static category = 'version'
+
+  static route = {
+    base: 'factorio-mod-portal/factorio-version',
+    pattern: ':modName',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Factorio Mod Portal factorio versions',
+      namedParams: { modName: 'rso-mod' },
+      staticPreview: this.render({ version: '0.14-1.1' }),
+      queryParams: {
+        range: 'range',
+      },
+    },
+  ]
+
+  static defaultBadgeData = { label: 'factorio version' }
+
+  static render({ version }) {
+    return { message: version, color: 'blue' }
+  }
+
+  transform({ earliest, latest }) {
+    let versions = ''
+    if (earliest === latest) {
+      versions = earliest
+    } else {
+      versions = `${earliest}-${latest}`
+    }
+    return versions
+  }
+
+  async handle({ modName }, queryParams) {
+    const { first_release, latest_release } = await this.fetch({ modName })
+    const range = queryParams.range !== undefined
+    const version = range
+      ? this.transform({
+          earliest: first_release.info_json.factorio_version,
+          latest: latest_release.info_json.factorio_version,
+        })
+      : latest_release.info_json.factorio_version
+    return this.constructor.render({ version })
+  }
+}
+
+// Badge for mod's last updated date
+class FactorioModPortalLastUpdated extends BaseFactorioModPortalService {
+  static category = 'activity'
+
+  static route = {
+    base: 'factorio-mod-portal/last-updated',
+    pattern: ':modName',
+  }
+
+  static examples = [
+    {
+      title: 'Factorio Mod Portal mod',
+      namedParams: { modName: 'rso-mod' },
+      staticPreview: this.render({
+        last_updated: new Date(),
+      }),
+    },
+  ]
+
+  static defaultBadgeData = { label: 'last updated' }
+
+  static render({ last_updated }) {
+    return {
+      message: formatDate(last_updated),
+      color: age(last_updated),
+    }
+  }
+
+  async handle({ modName }) {
+    const { latest_release } = await this.fetch({ modName })
+    return this.constructor.render({ last_updated: latest_release.released_at })
+  }
+}
+
+// Badge for mod's total download count
+class FactorioModPortalDownloads extends BaseFactorioModPortalService {
+  static category = 'downloads'
+
+  static route = {
+    base: 'factorio-mod-portal/downloads',
```

Comment:
Normally we use `/dt` as the route for a badge showing total downloads - see https://github.com/badges/shields/blob/master/doc/badge-urls.md